lib/backend/watchersyncer/watchercache.go

```diff
@@ -405,9 +408,11 @@ func (wc *watcherCache) markAsValid(resourceKey string) {
 	}
 }
 
-// sendWaitForDatastore signals to the syncer that this watcherCache is not in sync.  See finishResync()
+// sendWaitForDatastore signals to the syncer that this watcherCache is not in-sync if the number of consecutive errors
+// exceeds the error threshold.  See finishResync() for how the watcherCache goes back to in-sync.
 func (wc *watcherCache) sendWaitForDatastore() {
-	if wc.hasSynced {
+	wc.errors++
+	if wc.hasSynced && wc.errors > ErrorThreshold {
```

Comment:
should that be `>=`

Reply:
Don't think it makes a huge difference - 3 is sort of a magic number. Could just as well be 4.

Reply:
Looks like our retry interval is 1s right now, meaning that:
- one watch failure + one list failure
- one list failure
- one list failure

Will take 3 seconds total to hit 4 retries and trigger this behavior. I think we probably want to bump this number up - thinking something more like 15 seconds (retries) should count as a persistent error.